app/models/user.rb

```diff
@@ -64,6 +64,7 @@ def update_remote_user_info
 
   def destroy_remote_user_info_immediately
     RemoteUserInfo.new(self).destroy!
+    delete_data_held_in_account_api
```

Comment:
Put this behaviour in `RemoteUserInfo#destroy!`, as that's where the update logic is.